src/oscar/apps/communication/utils.py

```diff
@@ -156,17 +156,18 @@ def prepare_attachments(self, attachments):
 
         return content_attachments, file_attachments
 
-    def get_base_context(self):
+    def get_base_context(self, request=None):
         """
         Return context that is common to all emails
         """
-        return {'site': Site.objects.get_current()}
+        return {'site': Site.objects.get_current(request)}
 
     def get_messages(self, event_code, extra_context=None):
         """
         Return rendered messages
         """
-        context = self.get_base_context()
+        request = extra_context.get("request")
```

Comment:
This will throw an `AttributeError` if `extra_context` hasn't been supplied. I think also it would be better for `get_base_context` to return the request in its own return value.